targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/watchdog.c

```diff
@@ -0,0 +1,123 @@
+#include "watchdog_api.h"
+
+#include "reset_reason_api.h"
+
+#include "fsl_wdog.h"
+
+// Platform specific watchdog definitions
+#define LPO_CLOCK_FREQUENCY 1000
+#define MAX_PRESCALER       8
+#define MAX_TIMEOUT         0xFFFFFFFF
+
+// Number of decrements in the timeout register per millisecond
+#define TICKS_PER_MS (LPO_CLOCK_FREQUENCY / 1000)
+// Maximum timeout that can be specified in milliseconds
+#define MAX_TIMEOUT_MS ((MAX_TIMEOUT / TICKS_PER_MS) * MAX_PRESCALER)
+
+// Maximum supported watchdog timeout for given prescaler value
+#define CALCULATE_MAX_TIMEOUT_MS(scale) \
+  ((MAX_TIMEOUT / TICKS_PER_MS) * scale)
+
+
+#if defined(FSL_FEATURE_WDOG_HAS_WAITEN) && FSL_FEATURE_WDOG_HAS_WAITEN
+  #define PLATFORM_SUPPORTS_SLEEP true
+#else
+  #define PLATFORM_SUPPORTS_SLEEP false
+#endif
+
+
+static uint32_t calculate_prescaler_value(const uint32_t timeout_ms)
+{
+  if (timeout_ms > MAX_TIMEOUT_MS) {
+    return 0;
+  }
+
+  for (uint32_t scale = 1; scale < MAX_PRESCALER; ++scale) {
+    if (timeout_ms < CALCULATE_MAX_TIMEOUT_MS(scale)) {
+      return scale;
+    }
+  }
+
+  return 0;
+}
+
+
+watchdog_status_t hal_watchdog_init(const watchdog_config_t *config)
+{
+  // Validate the input parameters
+  if (config == NULL) {
+    return WATCHDOG_STATUS_INVALID_ARGUMENT;
+  }
+
+  if (config->timeout_ms > MAX_TIMEOUT_MS) {
+    return WATCHDOG_STATUS_INVALID_ARGUMENT;
+  }
+
+  if (config->window_ms > MAX_TIMEOUT_MS) {
+    return WATCHDOG_STATUS_INVALID_ARGUMENT;
+  }
+
+  if (config->window_ms > config->timeout_ms) {
+    return WATCHDOG_STATUS_INVALID_ARGUMENT;
+  }
+
+  if (config->enable_sleep && !PLATFORM_SUPPORTS_SLEEP) {
+    return WATCHDOG_STATUS_NOT_SUPPORTED;
+  }
+
+  wdog_config_t cfg;
+  cfg.enableWdog = true;
+  cfg.clockSource = kWDOG_LpoClockSource;
+  cfg.enableUpdate = true;
+  cfg.enableInterrupt = false;
+  cfg.enableWindowMode = config->enable_window;
+  cfg.workMode.enableWait = config->enable_sleep;
```

Comment:
This line should be dependent on either `#if defined(FSL_FEATURE_WDOG_HAS_WAITEN) && FSL_FEATURE_WDOG_HAS_WAITEN` or `#if PLATFORM_SUPPORTS_SLEEP`. Same as here:
https://github.com/ARMmbed/mbed-os/blob/4e222952d757e378d5a591ca7fd143dcda24fc7e/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/drivers/fsl_wdog.c#L67-L69